src/common/collision.cpp

```diff
@@ -0,0 +1,31 @@
+#include "collision.hpp"
+#include <iostream>
+
+void handle_circle_collision(std::vector<Circle*>& circles) {
```

Comment:
We might want this to be a static function on the Group class that actually looks like `Group.handleCollisions(std::vector<Group*>& groups)`, then you pass it a vector of groups and it knows that groups are represented by circles so it does the collision using the circle object of the group. that way you dont have to make a new vector for every single group every time you call this function

Reply:
Sure I'm down.

Reply:
Sure I'm down

Reply:
Done in 9d5a236